File: Patrones/Bridge/C++/BridgeStackSimple/src/StackFIFO.cpp
#include <StackFIFO.h>

// way to construct a super object
StackFIFO::StackFIFO() : Stack("array"){
	temp = new StackList();
}

StackFIFO::StackFIFO(string s) : Stack(s){}

int StackFIFO::pop(){
   
    while ( !isEmpty()) {
        temp->push( pop() );
    }

    int ret =  temp->pop();
    while ( !temp->isEmpty()) {
        push( temp->pop() );
    }
    
    return ret;
}

StackFIFO::~StackFIFO(){
	delete temp;
	temp = 0;
}

File: Patrones/Bridge/Java/BridgeStackExtended/src/example/StackFIFO.java
package example;

public class StackFIFO extends Stack {
    
	private StackImpl temp = new StackList();
    
    public StackFIFO() {
        super( "array" );
    }
    
    public StackFIFO( String s ) {
        super( s );
    }
    
    public int pop() {
        while ( ! isEmpty()) {
            temp.push( super.pop() );
        }
        int ret =  temp.pop();
        while ( ! temp.isEmpty()) {
            push( temp.pop() );
        }
        return ret;
    }
}
